Clover/app/src/main/java/org/floens/chan/core/site/sites/dvach/DvachApi.java

```diff
@@ -0,0 +1,238 @@
+package org.floens.chan.core.site.sites.dvach;
+
+import android.util.JsonReader;
+
+import org.floens.chan.core.model.Post;
+import org.floens.chan.core.model.PostImage;
+import org.floens.chan.core.site.SiteEndpoints;
+import org.floens.chan.core.site.common.CommonSite;
+import org.floens.chan.core.site.parser.ChanReaderProcessingQueue;
+import org.jsoup.parser.Parser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.floens.chan.core.site.SiteEndpoints.makeArgument;
+
+public class DvachApi extends CommonSite.CommonApi {
+    DvachApi(CommonSite commonSite) {
+        super(commonSite);
+    }
+
+    @Override
+    public void loadThread(JsonReader reader, ChanReaderProcessingQueue queue) throws Exception {
+        reader.beginObject(); // Main object
+
+        while (reader.hasNext()) {
+
+            if (reader.nextName().equals("threads")) {
+                reader.beginArray(); // Threads array
+
+                while (reader.hasNext()) {
+                    reader.beginObject(); // Posts object
+                    if (reader.nextName().equals("posts")) {
+                        reader.beginArray(); // Posts array
+                        while (reader.hasNext()) {
+                            readPostObject(reader, queue);
+                        }
+                        reader.endArray();
+                    }
+                    reader.endObject();
+                }
+
+                reader.endArray();
+            } else {
+                reader.skipValue();
+            }
+        }
+
+        reader.endObject();
+    }
+
+    @Override
+    public void loadCatalog(JsonReader reader, ChanReaderProcessingQueue queue) throws Exception {
+        reader.beginObject(); // Main object
+
+        while (reader.hasNext()) {
+            if (reader.nextName().equals("threads")) {
+                reader.beginArray(); // Threads array
+
+                while (reader.hasNext()) {
+                    readPostObject(reader, queue);
+                }
+
+                reader.endArray();
+            } else {
+                reader.skipValue();
+            }
+        }
+
+        reader.endObject();
+    }
+
+    @Override
+    public void readPostObject(JsonReader reader, ChanReaderProcessingQueue queue) throws Exception {
+        Post.Builder builder = new Post.Builder();
+        builder.board(queue.getLoadable().board);
+
+        SiteEndpoints endpoints = queue.getLoadable().getSite().endpoints();
+
+        List<PostImage> files = new ArrayList<>();
+
+        reader.beginObject();
+        while (reader.hasNext()) {
+            String key = reader.nextName();
+
+            switch (key) {
+                case "no":
+                    builder.id(Integer.parseInt(reader.nextString()));
```

Comment:
I see builder.id being used in both "no" and "num". Why is that?

Reply:
2ch api doesn't have "no" fields, it can be removed